src/matcha_ml/config/matcha_config.py

```diff
@@ -0,0 +1,206 @@
+            }
+
+        return state_dictionary
+
+    @staticmethod
+    def from_dict(state_dict: Dict[str, Dict[str, str]]) -> "MatchaConfig":
+        """A function to convert a dictionary representation of the Matcha config file into a MatchaConfig instance.
+
+        Args:
+            state_dict (Dict[str, Dict[str, str]]): the dictionary representation of the Matcha config file.
+
+        Returns:
+            MatchaConfig: the MatchaConfig representation of the MatchaConfig instance.
+        """
+        components: List[MatchaConfigComponent] = []
+        for resource, properties in state_dict.items():
+            components.append(
+                MatchaConfigComponent(
+                    name=resource,
+                    properties=[
+                        MatchaConfigComponentProperty(name=key, value=value)
+                        for key, value in properties.items()
+                    ],
+                )
+            )
+
+        return MatchaConfig(components=components)
+
+
+class MatchaConfigService:
+    """A service for handling the Matcha config file."""
+
+    @staticmethod
+    def write_matcha_config(matcha_config: MatchaConfig) -> None:
+        """A function for writing the local Matcha config file.
+
+        Args:
+            matcha_config (MatchaConfig): the MatchaConfig representation of the MatchaConfig instance.
+        """
+        local_config_file = os.path.join(os.getcwd(), DEFAULT_CONFIG_NAME)
+
+        with open(local_config_file, "w") as file:
+            json.dump(matcha_config.to_dict(), file)
+
+    @staticmethod
+    def read_matcha_config() -> MatchaConfig:
+        """A function for reading the Matcha config file into a MatchaConfig object.
+
+        Returns:
+           MatchaConfig: the MatchaConfig representation of the MatchaConfig instance.
+
+        Raises:
+            MatchaError: raises a MatchaError if the local config file could not be read.
+        """
+        local_config_file = os.path.join(os.getcwd(), DEFAULT_CONFIG_NAME)
+
+        if os.path.exists(local_config_file):
+            with open(local_config_file) as config:
+                local_config = json.load(config)
+
+            return MatchaConfig.from_dict(local_config)
+        else:
+            raise MatchaError(
+                f"No '{DEFAULT_CONFIG_NAME}' file found, please generate one by running 'matcha provision', or add an existing ''{DEFAULT_CONFIG_NAME}'' file to the root project directory."
+            )
+
+    @staticmethod
+    def config_file_exists() -> bool:
+        """A convencience function which checks for the existence of the matcha.config.json file.
+
+        Returns:
+            True if the matcha.config.json file exists, False otherwise.
+        """
+        return os.path.exists(os.path.join(os.getcwd(), DEFAULT_CONFIG_NAME))
+
+    @staticmethod
+    def update(components: Union[MatchaConfigComponent, List[MatchaConfigComponent]]) -> None:
```

Comment:
There doesn't appear to be any tests for this function?

Reply:
Added